### PR TITLE
ENH: fixes to Signal from working at TES

### PR DESCRIPTION
**Pull request by tacaswell**
- the check in set_and_wait used to fail with
   value error for arrays and the exception was going
   to the debug log
 - at TES we used list(...) to get around this
 - Teddy updated Signal to correctly report it's type,
   but the type sniffer did not correctly handle lists

### CODE REVIEW

**Comment by mrakitin** [comment]
Any ideas on having a test for this case?

**Comment by tacaswell** [comment]
Yes, I just wanted to get the code out of my head before I left TES.

**Comment by mrakitin** [comment]
That's where we saw the problem: https://github.com/NSLS-II-TES/profile_collection/commit/cf96189fe8d6dacfae507ed0d617a8b093f8d14a#diff-c675cd8248c0fa1ec9653e62d8ec1ccfL189

And that's where it was temporary fixed: https://github.com/NSLS-II-TES/profile_collection/commit/cf96189fe8d6dacfae507ed0d617a8b093f8d14a#diff-c675cd8248c0fa1ec9653e62d8ec1ccfR143

**Comment by klauer** [comment]
Looks like this is an important one to get in?

File: ophyd/signal.py [hunk 1]
```diff
@@ -164,11 +164,12 @@ def set_thread():
                                value)
                 success = False
             except Exception as ex:
-                self.log.debug('set_and_wait(%r, %s) failed', self.name, value,
-                               exc_info=ex)
+                self.log.exception('set_and_wait(%r, %s) failed',
```
Comment (klauer) [inline ophyd/signal.py]: This turns the `debug` message into an `error` message. Was that intentional or was the desire to just remove `exc_info=ex`?

**Comment by klauer** [comment]
Wrapping the important `data_type` bits into #653.

**Comment by danielballan** [comment]
Thank you, @klauer.

File: ophyd/signal.py (hunk 1 above)
Comment (tacaswell) [reply ophyd/signal.py]: Mostly driven by number 2, but I think number 1 is also a correct change as `set_and_wait` failing should be a big deal.

File: ophyd/utils/epics_pvs.py [hunk 2]
```diff
@@ -282,11 +282,15 @@ def _compare_maybe_enum(a, b, enums, atol, rtol):
...
+    ret = a == b
+    try:
+        return bool(ret)
+    except ValueError:
+        return np.all(ret)
 
 
 _type_map = {'number': (float, np.floating),
-             'array': (np.ndarray, ),
+             'array': (np.ndarray, list, tuple),
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply ophyd/utils/epics_pvs.py]: we also require it to have a concrete length.  Agree a refactor is a good idea, but I think out of scope for current effort.

**Pull request merged**